Sources/NIOFoundationCompat/ByteBuffer-foundation.swift

```diff
@@ -265,6 +265,101 @@ extension ByteBuffer {
         }
         return written
     }
+
+    // MARK: - UUID
+
+    /// Get a `UUID` from the 16 bytes starting at `index`. This will not change the reader index.
+    /// If there are less than 16 bytes starting at `index` then `nil` will be returned.
+    ///
+    /// - Parameters:
+    ///   - index: The starting index of the bytes of interest into the `ByteBuffer`.
+    /// - Returns: A `UUID` value containing the bytes of interest or `nil` if the selected bytes
+    ///            are not readable or there were not enough bytes.
+    public func getUUID(at index: Int) -> UUID? {
+        guard let chunk1 = self.getInteger(at: index, as: UInt64.self),
+              let chunk2 = self.getInteger(at: index + 8, as: UInt64.self) else {
+            return nil
+        }
+
+        let uuidBytes = (
+            UInt8(truncatingIfNeeded: chunk1 >> 56),
+            UInt8(truncatingIfNeeded: chunk1 >> 48),
+            UInt8(truncatingIfNeeded: chunk1 >> 40),
+            UInt8(truncatingIfNeeded: chunk1 >> 32),
+            UInt8(truncatingIfNeeded: chunk1 >> 24),
+            UInt8(truncatingIfNeeded: chunk1 >> 16),
+            UInt8(truncatingIfNeeded: chunk1 >> 8),
+            UInt8(truncatingIfNeeded: chunk1),
+            UInt8(truncatingIfNeeded: chunk2 >> 56),
+            UInt8(truncatingIfNeeded: chunk2 >> 48),
+            UInt8(truncatingIfNeeded: chunk2 >> 40),
+            UInt8(truncatingIfNeeded: chunk2 >> 32),
+            UInt8(truncatingIfNeeded: chunk2 >> 24),
+            UInt8(truncatingIfNeeded: chunk2 >> 16),
+            UInt8(truncatingIfNeeded: chunk2 >> 8),
+            UInt8(truncatingIfNeeded: chunk2)
+        )
```

Comment:
As we have to copy exactly 16bytes here, I think I would go for using `unsafe` APIs.

Reply:
My gut feeling is that this codepath isn't common enough to warrant dropping into unsafe land and the current approach shouldn't perform terribly.

Reply:
[It totally does](https://swift.godbolt.org/z/MvPTY4efv):

```
        mov     rdx, rsi
        mov     rax, rdi
        bswap   rdx
        bswap   rax
        ret
```